test/contracts/DefiWithMockPriceFeed.test.ts

```diff
@@ -1,42 +1,66 @@
-import { ethers } from "hardhat";
+import {ethers} from "hardhat";
 import {Wallet} from "ethers";
 import chai from "chai";
-import { solidity } from "ethereum-waffle";
-import { MockDefi } from "../../typechain/MockDefi";
-import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
-import { syncTime } from "../_helpers";
-const { wrapContract } = require("../../utils/contract-wrapper");
+import {solidity} from "ethereum-waffle";
+import {MockDefi} from "../../typechain/MockDefi";
+import {SignerWithAddress} from "@nomiclabs/hardhat-ethers/signers";
+import {mockModule, syncTime} from "../_helpers";
 
 import {PriceFeedWithClearing} from "../../typechain/PriceFeedWithClearing";
+import {wrapContract} from "../../utils/contract-wrapper";
+import * as MockProvider from "../../utils/mock-price-package";
+import * as sinon from 'sinon';
+
 chai.use(solidity);
 
-const { expect } = chai;
+const {expect} = chai;
 
 const toBytes32 = ethers.utils.formatBytes32String;
 const serialized = function (x: number): number {
-    return x * 10**8;
+  return x * 10 ** 8;
 };
 
-describe("MockDefi with Proxy contract and mock pricing Data", function() {
+describe("MockDefi with Proxy contract and mock pricing Data", function () {
 
   const PRIV = "0xae2b81c1fe9e3b01f060362f03abd0c80a6447cfe00ff7fc7fcf000000000000";
 
+  let sandbox: sinon.SinonSandbox;
+
+  const mockProvider = mockModule<typeof MockProvider>(MockProvider, {
```

Comment:
`mockProvider` is a function that takes two parameters - sandbox and overridden mock value. If not overridden by certain test, default value (configured here) will be used.